src/main/java/org/opensearch/knn/index/KNNMethod.java

```diff
@@ -77,6 +78,35 @@ public ValidationException validate(KNNMethodContext knnMethodContext) {
         return validationException;
     }
 
+    public ValidationException validateWithData(KNNMethodContext knnMethodContext, TrainingDataSpec trainingDataSpec) {
```

Comment:
can we add documentation for public methods?

Reply:
Good call, I meant to do that at the end and forgot about it